compiler/frontend/src/org/jetbrains/kotlin/psi/psiUtil/ktPsiUtil.kt

```diff
@@ -454,3 +454,17 @@ fun checkReservedPrefixWord(sink: DiagnosticSink, element: PsiElement, word: Str
     }
 }
 
+fun KtNameReferenceExpression.getPrimaryConstructorParameterWithSameName(): KtParameter? {
+    return getPrimaryConstructorParameterWithName(getReferencedName())
+}
+
+private fun KtElement.getPrimaryConstructorParameterWithName(name: String): KtParameter? {
```

Comment:
What would be a good name for this sequence method? `getNonStaticOuterClassSequence()`?